File: day1/1.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

// func calculateDepthWindow(i int) {
//     return i
// }
func main() {
	fmt.Println("Solving!")
	input, err := os.ReadFile("/Users/gyao/Downloads/1.txt")
	check(err)
	result := 0
	depths := strings.Split(string(input), "\n")
	var last_depth_window int
	for i := 0; i < 3; i++ {
		depth, err := strconv.Atoi(depths[i])
		check(err)
		last_depth_window += depth
	}

	for i := 1; i < len(depths)-3; i++ {
		var depth_window int
		for j := i; j < i+3; j++ {
			depth, err := strconv.Atoi(depths[j])
			check(err)
			depth_window += depth
		}
		if depth_window > last_depth_window {
			result += 1
		}
		last_depth_window = depth_window
	}
	fmt.Println(result)
}

//
// func main() {
//     fmt.Println("Solving!")
//     input, err := os.ReadFile("/Users/gyao/Downloads/1.txt")
//     check(err)
//     result := 0
//     depths := strings.Split(string(input),"\n")
//     last_depth, err := strconv.Atoi(depths[0])
//     check(err)
//     for index, depth := range depths {
//       if index == 0 || len(depths) == 0 {
//         continue
//       }
//       d, err := strconv.Atoi(depth)
//       fmt.Println(d)
//       if err != nil || d == 0{
//         fmt.Println("Empty value found, skipping")
//       }
//       if d > last_depth {
//         result += 1
//       }
//       last_depth = d
//     }
//   fmt.Println(result)
// }

File: day1/1.py
import io
class Result:
    @staticmethod
    def reader(filename):
        data = []
        with open(filename, "r") as f:
            for line in f:
                data.append(line.splitlines()[0])
        return data
class day1(Result):
    input = Result.reader("/Users/gyao/Downloads/1.txt")
    data = [int(i) for i in input]
    # def depth_counter(data):
    #     depth_counter = 0
    #     for i in range(1,len(data)):
    #         if data[i] > data[i-1]:
    #             depth_counter += 1
    #     return depth_counter
    def depth_counter(data):
        result = 0
        last_depth_window = data[0] + data[1] + data[2]
        for i in range(1,len(data)-2):
            depth_window = data[i] + data[i+1] + data[i+2]
            if depth_window > last_depth_window:
                result += 1
            last_depth_window = depth_window
        return result
    result = depth_counter(data)
    print(result)
